File: 2-Patron/php/Criatura.php
<?php
abstract class Criatura
{
  private $nombre;
  private $defensa;
  private $ataque;
  private $nivel;

  function __construct($nombre, $defensa, $ataque, $nivel)
  {
    $this->nombre = $nombre;
    $this->defensa = $defensa;
    $this->ataque = $ataque;
    $this->nivel = $nivel;
  }

  function getNombre()
  {
    return $this->nombre;
  }

  function setNombre($nNombre)
  {
    $this->nombre = $nNombre;
  }

  function getDefensa()
  {
    return $this->defensa;
  }

  function setDefensa($nDefensa)
  {
    $this->$defensa = $nDefensa;
  }

  function getAtaque()
  {
    return $this->ataque;
  }

  function setAtaque($nAtaque)
  {
    $this->ataque = $nAtaque;
  }

  function getNivel()
  {
    return $this->nivel;
  }

  function setNivel($nNivel)
  {
    $this->nivel = $nNivel;
  }

  function verStats()
  {
    return  "Nombre: " . $this->nombre .
            "Defensa: " . $this->defensa .
            "Ataque: " $this->ataque .
            "Nivel" $this->nivel;
  }

  abstract function decSalud();
  abstract function incNivel();
}
?>

File: 2-Patron/Java/Criatura.java
package abstractmasterwhole;

    public abstract class Criatura{
    private String nombre;
    private int defensa;
    private int ataque;
    private int nivel;
    public Criatura(String nombre, int defensa, int ataque, int nivel){
        this.nombre = nombre;
        this.defensa = defensa;
        this.ataque = ataque;
        this.nivel = nivel;
    }
    public String getNombre(){
        return this.nombre;
    }
    public void setNombre(String nombre){
        this.nombre = nombre;
    }
    public int getDefensa(){
        return this.defensa;
    }
    public void setDefensa(int defensa){
        this.defensa = defensa;
    }
    public int getAtaque(){
        return this.ataque;
    }
    public void setAtaque(int ataque){
        this.ataque = ataque;
    }
    public int getNivel(){
        return this.nivel;
    }
    public void setNivel(int nivel){
        this.nivel = nivel;
    }
    
    abstract public int decSalud(int dec);
    abstract public  int incNivel(); 
    
    public String verStats(){
        return ("Nombre: "+ this.nombre+"\n Defensa: "+this.defensa+ "\n Ataque: "
                +  this.ataque + "\n Nivel: "+ this.nivel);
    }
        
}
